celery/app/base.py

```diff
@@ -423,6 +427,18 @@ def __autoset(self, key, value):
             self._preconf[key] = value
             self._preconf_set_by_auto.add(key)
 
+    def _validate_broker_url(self, url):
+        if not url:
+            return
+        try:
+            parse_url(url)
+        except ValueError as e:
+            raise ImproperlyConfigured(
+                f"Invalid broker URL: {url}, {e}.\n\n"
+                "Special characters in credentials must be percent-encoded.\n"
```

Comment:
"userinfo" is the term used in the RFCs (ex. [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1))
```suggestion
                "Special characters in userinfo must be percent-encoded.\n"
```